src/App.tsx

```diff
@@ -1,27 +1,65 @@
-import React from 'react';
-import './App.scss';
+import React, { useState } from 'react';
+import { Good } from './types/Good';
 import { GoodsList } from './GoodsList';
+import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={() => {
+          getAll()
+            .then((data) => {
+              setGoods(data);
+            }).catch(() => {
+              throw new Error('Error, sorry :(');
+            });
```

Comment:
let's move it to the separate method
Also, would be great to use async & await instead of then